daemon/api.go

```diff
@@ -1316,3 +1323,42 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 
 	return SyncResponse(createResponseData, nil)
 }
+
+type buyResponseData struct {
+	State string `json:"state,omitempty"`
+}
+
+func postBuy(c *Command, r *http.Request, user *auth.UserState) Response {
+	var buyInstruction store.BuyOptions
+
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&buyInstruction)
+	if err != nil {
+		return BadRequest("cannot decode buying instruction data from request body: %v", err)
+	}
+
+	buyInstruction.Auther, err = c.d.auther(r)
```

Comment:
`c.d.auther` is obsolete.. we should really remove it.

We have the authenticated user at hand here:  

```
if user != nil {
        opts.Auther = user.Authenticator()
}
```